src/components/datagrid/_data_grid.scss

```diff
@@ -97,7 +97,7 @@
 .euiDataGrid__overflow {
   overflow-y: hidden;
   height: 100%;
-  background: $euiPageBackgroundColor;
+  background: $euiColorEmptyShade;
```

Comment:
@snide please let me know if you have any objections to this change. Couldn't figure out why you were using `$euiPageBackgroundColor` but it was overriding the styles for footer/header (i.e. always giving them a background).